text/unstable_strip.ts

```diff
@@ -0,0 +1,161 @@
+  str: string,
+  pattern: string | RegExp,
+  options?: StripOptions,
+): string {
+  const { source, flags } = cloneAsStatelessRegExp(pattern);
+  return stripByRegExp(
+    stripByRegExp(str, new RegExp(`^${source}`, flags), options),
+    new RegExp(`${source}$`, flags),
+    options,
+  );
+}
+
+/**
+ * Strips the specified pattern from the start of the string.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param str The input string
+ * @param pattern The pattern to strip from the input
+ * @param options Options for the strip operation
+ *
+ * @example Strip using a string
+ * ```ts
+ * import { stripStart } from "@std/text/unstable-strip";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(stripStart("---x---", "-"), "x---");
+ * ```
+ *
+ * @example Strip using a regexp
+ * ```ts
+ * import { stripStart } from "@std/text/unstable-strip";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(stripStart("-_-x-_-", /[-_]/), "x-_-");
+ * ```
+ *
+ * @example Strip a specified count
+ * ```ts
+ * import { stripStart } from "@std/text/unstable-strip";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(stripStart("---x---", "-", { count: 1 }), "--x---");
+ * ```
+ */
+export function stripStart(
+  str: string,
+  pattern: string | RegExp,
+  options?: StripOptions,
+): string {
+  const { source, flags } = cloneAsStatelessRegExp(pattern);
+  return stripByRegExp(str, new RegExp(`^${source}`, flags), options);
+}
+
+/**
+/**
+ * Strips the specified pattern from the start of the string.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param str The input string
+ * @param pattern The pattern to strip from the input
+ * @param options Options for the strip operation
+ *
+ * @example Strip using a string
+ * ```ts
+ * import { stripEnd } from "@std/text/unstable-strip";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(stripEnd("---x---", "-"), "---x");
+ * ```
+ *
+ * @example Strip using a regexp
+ * ```ts
+ * import { stripEnd } from "@std/text/unstable-strip";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(stripEnd("-_-x-_-", /[-_]/), "-_-x");
+ * ```
+ *
+ * @example Strip a specified count
+ * ```ts
+ * import { stripEnd } from "@std/text/unstable-strip";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(stripEnd("---x---", "-", { count: 1 }), "---x--");
+ * ```
+ */
+export function stripEnd(
+  str: string,
+  pattern: string | RegExp,
+  options?: StripOptions,
+): string {
+  const { source, flags } = cloneAsStatelessRegExp(pattern);
+  return stripByRegExp(str, new RegExp(`${source}$`, flags), options);
+}
+
+function stripByRegExp(
+  str: string,
+  regExp: RegExp,
+  options?: StripOptions,
+): string {
+  let prev = str;
+
+  const count = options?.count ?? Infinity;
+
+  for (let i = 0; i < count; ++i) {
+    str = str.replace(regExp, "");
+    if (str === prev) break;
+    prev = str;
+  }
```

Comment:
Why `Infinity`? If the string doesn't include the pattern to strip, it will loop forever, won't it?

Reply:
> Why `Infinity`? If the string doesn't include the pattern to strip, it will loop forever, won't it?

It'll break on the `str === prev` condition as soon as there's nothing left to strip.

```ts
const aaa = 'a'.repeat(1000)
const aab = aaa + 'b'
const bbb = 'b'.repeat(1000)

stripStart(aaa, 'a') === ''
stripStart(aab, 'a') === 'b'
stripStart(bbb, 'a') === bbb
```